### PR TITLE
build(client): use fluid-tsc for Node16 test builds

### PR DESCRIPTION
**Pull request by jason-ha**
build(client): use fluid-tsc for Node16 test builds
- remove now stale tsc-multi configurations and replace all use with fluid-tsc.
- enable ESM build for tree test
  - Make ajvValidator CommonJS to accommodate typescript's rejection of Ajv's handcrafted ES Module support.
  - *However*, ESM test build is not runnable per internal module import of id-compressor CommonJS output (`@fluidframework/id-compressor/dist/test`).
- correct @fluid-private/test-end-to-end-tests build dependencies (was changed to ESNext in #15976) and @fluidframework/container-runtime test support was previously moved to production (in #18826).
- update build-tools

Updated the following:

  client (release group)
  client-release-group-root

Dependencies on build-tools updated:

  @fluid-tools/build-cli: 0.34.0
  @fluidframework/build-tools: 0.34.0
  @fluidframework/bundle-size-tools: 0.34.0
  @fluid-tools/version-tools: 0.34.0

Command used:

```shell
pnpm flub bump deps build-tools -g client -t latest --updateChecker homegrown -x
```

### CODE REVIEW

**Review by DLehenbauer — APPROVED** [review]
It's lovely.  Thank you, Jason!

File: fluidBuild.config.cjs [hunk 1]
```diff
@@ -60,6 +60,8 @@ module.exports = {
 			"api-extractor:commonjs",
 			"api-extractor:esnext",
 		],
+		"build:test:cjs": ["typetests:gen", "tsc", "api-extractor:commonjs"],
+		"build:test:esm": ["typetests:gen", "build:esnext", "api-extractor:esnext"],
```
Comment (jason-ha) [inline fluidBuild.config.cjs:64]: @tylerbutler, if a package doesn't have a `tsc` or `build:exnext` script would the general dependencies of `tsc` on `^tsc` and `build:esnext` on `^build:esnext` be transitively picked up? Or would they get ignored?

Comment (tylerbutler) [reply fluidBuild.config.cjs:64]: They'd be ignored. The dependency "chain" is broken in that case so unless another task brought those tasks in, they'd be omitted. 

@curtisman did I get that right?

Comment (curtisman) [reply fluidBuild.config.cjs:64]: Yes, the relevent code is in `buildGraph.ts`: `BuildPackage.getMatchedTasks`

**Pull request merged**